miqcli/cli/main.py

```diff
@@ -228,6 +234,13 @@ def invoke(self, ctx):
         :param ctx: Click context.
         :type ctx: Namespace
         """
+        # first lets make sure the sub-command given is valid
+        if ctx.protected_args[0] not in self.list_commands(ctx):
+            # invalid command, display list of available commands/exit
+            display_commands(ctx)
+            log.abort('Command "{0}" is invalid. Please choose a valid command'
+                      ' from the list above.'.format((ctx.protected_args[0])))
```

Comment:
This is a copy of the same code line line 90, and both use the same args. Is there value in consolidating this into a single function for invalid commands somewhere?
```python
def _abort_invalid_command(ctx, name):
    display_commands(ctx)
    log.abort('Command "{0}" is invalid. Please choose a valid command'
                   'from the list above.'.format((name)))
```

Reply:
@seandst +1. Will make the change and amend shortly. Allows other modules to use the function and reduces duplication.